_posts/2019-10-11-stripe-connect-in-rails-part-2.md

```diff
@@ -0,0 +1,164 @@
+---
+layout: post
+title: "Implementing Stripe Connect in Rails: Part 2"
+date: 2019-10-11 14:00:00
+categories: ["rails", "stripe"]
+author: "luciano"
+---
+
+Some time ago we wrote [an article](https://www.ombulabs.com/blog/rails/stripe/stripe-connect-part-1.html) to explain how to implement [Stripe Connect](https://stripe.com/connect) in a [Rails](https://rubyonrails.org) application. That article covered mainly the connection part between Rails and Stripe. This one will cover the fun part, which is making transactions, charges, refunds, and more.
+
+If you haven't check the [first part](https://www.ombulabs.com/blog/rails/stripe/stripe-connect-part-1.html) yet I recommend you to do it since this article will continue from where that one ended.
+
+<!--more-->
+
+## Frontend
+
+Before diving into the Stripe API, one of the first things that you should implement is a form that captures the payment information (e.g. credit card).
+The simplest way to do this is using [Stripe Checkout](https://stripe.com/docs/payments/checkout), but if you want a more customizable version you can use [Stripe Elements](https://stripe.com/docs/web/setup).
+I recommend you to follow the official documentation of any these options so you have everything ready to continue with the backend work.
+
+## Backend
+
+### Charges
+
+Once you have your form ready to collect payment information you'll probably want it to execute a [Charge](https://stripe.com/docs/api/charges) when it's submitted.
+
+```ruby
+# app/controllers/payments_controller.rb
+
+class PaymentsController < ApplicationController
+  # triggered after submit payment form
+  def create
+    # ...
+
+    charge = StripeChargeService.charge!(params[:stripeToken])
+
+    # ... (create Payment, store charge id in db, respond, etc.)
+  end
+end
+```
+There are a few different [approaches](https://stripe.com/docs/connect/charges#choosing-approach) to implement the Stripe Charge. Each implementation fits better for different needs since in some cases the platform account should be the one that pays for the [Stripe fee](https://stripe.com/connect/pricing) and in some other cases it would be the connected account. I recommend you to take a look at the [official guides](https://stripe.com/docs/connect/charges) to make sure that you choose the one that fits better for you.
```

Comment:
```suggestion
There are a few different [approaches](https://stripe.com/docs/connect/charges#choosing-approach) to implement the Stripe Charge. Each implementation fits different needs, since in some cases the platform account should be the one that pays for the [Stripe fee](https://stripe.com/connect/pricing) and in some other cases it would be the connected account. I recommend that you take a look at the [official guides](https://stripe.com/docs/connect/charges) to make sure that you choose the one that fits better for you.
```